assets/js/data/cart/tracking.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * External dependencies
+ */
+import { getSetting } from '@woocommerce/settings';
+import { Cart, isObject, objectHasProp } from '@woocommerce/types';
+
+let cartPageViewTracked = false;
+let checkoutPageViewTracked = false;
+
+interface StorePageDetails {
+	id: number;
+	title: string;
+	permalink: string;
+}
+
+interface StorePages {
+	checkout: StorePageDetails;
+	cart: StorePageDetails;
+	myaccount: StorePageDetails;
+	privacy: StorePageDetails;
+	shop: StorePageDetails;
+	terms: StorePageDetails;
+}
+export const maybeTrackCheckoutPageView = ( cart: Cart ) => {
+	if ( checkoutPageViewTracked ) {
+		return;
+	}
+	checkoutPageViewTracked = true;
+	const storePages = getSetting< StorePages >( 'storePages', {} );
+	if ( ! objectHasProp( storePages, 'checkout' ) ) {
+		return;
+	}
+	if ( storePages?.checkout?.permalink !== window.location.href ) {
+		return;
+	}
```

Comment:
This works fine with trailing slashes and query params? I don't think Checkout has query params now but it might have in the future with "continue to pay" flows.